app: start init script event loop explicitly (#9264)
tarantool/tarantool: changelogs/unreleased/gh-9266-fix-on-shutdown-and-osexit-from-cmd-expr.md

## bugfix/box

> * Fixed a bug when on shutdown trigger is not run if "os.exit()" is called
>   from -e command line option (gh-9266).

Reviewer: ```suggestion
* Fixed a bug when `on_shutdown` triggers weren't run if `os.exit()` was
  called from `-e` command-line option (gh-9266).
```

Author: Done.
